docs/dev/ci.rst

```diff
@@ -0,0 +1,42 @@
+Continuous Integration
+======================
+
+Almost everything that is run in CI (using CircleCI_) is designed to be possible to run locally.
+This helps in that it's easier to use your own laptop to debug potential problems that only
+happen in CI.
+
+.. note:: "Almost everything", because pushing to `Docker Hub`_ is not something you can do
+          from your laptop.
+
+
+.. _Docker Hub: https://hub.docker.com/
+.. _CircleCI: https://circleci.com/gh/mozilla/normandy
+
+
+Running Suite
+-------------
+
+The steps in ``.circleci/config.yml`` should be sufficiently self-explanatory. Each step should
+work to just copy and paste into your own terminal.
+
+There are some exceptions though. For example, when a step is set up to run with
+``background: true`` it means you can run it in its own dedicated terminal and run the
+other commands in another terminal.
+
+Also, steps with ``when: always`` means they can be run after any other step. In particular
+we use this to extract content from containers that have finished successfully or failed.
```

Comment:
I think "they can be run after any other step" is confusing. It makes me think that the step in question is somehow run multiple times after every other step, or something else. I think that the comment in the CircleCI config next to `when: always` is sufficient, and we don't need to mention it here.